BEST_PRACTICES.md

```diff
@@ -550,7 +550,7 @@ Additional capabilities of classes:
 
 ### Use of Self
 
-Always use `self` when referencing properties. It will make your life simpler. Trust us.
+We don’t write `self` explicitly, because Swift assumes that you are referring to a property or method of the current instance whenever you use a known property or method name within a method.
```

Comment:
Change this to `within a class` maybe? 

And I'd like to add (something like) this:
> for the team it was a choice between: 
- not using `self` for the reason that it removes unnecessary verboseness and is the swift guidelines from apple 
- use `self` for code clarity reasons.